comixed-state/src/main/java/org/comixedproject/state/comic/ComicStateHandler.java

```diff
@@ -0,0 +1,108 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses>
+ */
+
+package org.comixedproject.state.comic;
+
+import static org.comixedproject.state.comic.ComicStateConstants.HEADER_COMIC;
+
+import java.util.HashSet;
+import java.util.Set;
+import lombok.extern.log4j.Log4j2;
+import org.comixedproject.model.comic.Comic;
+import org.comixedproject.model.comic.ComicState;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.support.MessageBuilder;
+import org.springframework.statemachine.StateMachine;
+import org.springframework.statemachine.access.StateMachineAccess;
+import org.springframework.statemachine.access.StateMachineFunction;
+import org.springframework.statemachine.state.State;
+import org.springframework.statemachine.support.DefaultStateMachineContext;
+import org.springframework.statemachine.support.LifecycleObjectSupport;
+import org.springframework.statemachine.support.StateMachineInterceptorAdapter;
+import org.springframework.statemachine.transition.Transition;
+import org.springframework.stereotype.Component;
+
+/**
+ * <code>ComicStateHandler</code> handles firing actions and notifying listeners for state changes
+ * on {@link Comic} instances.
+ *
+ * @author Darryl L. Pierce
+ */
+@Component
+@Log4j2
+public class ComicStateHandler extends LifecycleObjectSupport {
+  @Autowired private StateMachine<ComicState, ComicEvent> stateMachine;
+
+  private Set<ComicStateChangeListener> listeners = new HashSet<>();
+
+  @Override
+  protected void onInit() throws Exception {
+    this.stateMachine
+        .getStateMachineAccessor()
+        .doWithAllRegions(
+            new StateMachineFunction<StateMachineAccess<ComicState, ComicEvent>>() {
+              @Override
+              public void apply(final StateMachineAccess<ComicState, ComicEvent> function) {
+                function.addStateMachineInterceptor(
+                    new StateMachineInterceptorAdapter<>() {
+                      @Override
+                      public void postStateChange(
+                          final State<ComicState, ComicEvent> state,
+                          final Message<ComicEvent> message,
+                          final Transition<ComicState, ComicEvent> transition,
+                          final StateMachine<ComicState, ComicEvent> stateMachine,
+                          final StateMachine<ComicState, ComicEvent> rootStateMachine) {
+                        listeners.forEach(listener -> listener.onComicStateChange(state, message));
+                      }
+                    });
+              }
```

Comment:
Couldn't we replace this code with something like this: 
`
   this.stateMachine.getStateMachineAccessor().doWithAllRegions(

                function -> function.addStateMachineInterceptor(

                    new StateMachineInterceptorAdapter<>() {

                      @Override
                      public void postStateChange(

                          final State<ComicState, ComicEvent> state,
                          final Message<ComicEvent> message,
                          final Transition<ComicState, ComicEvent> transition,
                          final StateMachine<ComicState, ComicEvent> stateMachine,
                          final StateMachine<ComicState, ComicEvent> rootStateMachine) {
                        listeners.forEach(listener -> listener.onComicStateChange(state, message));

                      }
                    }));`

Reply:
Definitely can. I'm still adapting to the newer (to me) Java syntax and appreciate any point where you can point these out to me. 👍